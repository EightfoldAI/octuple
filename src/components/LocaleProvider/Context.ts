import { createContext } from 'react';
import type { Locale } from '.';

export type LocaleContextProps = Partial<Locale> & { exist?: boolean };

const LocaleContext = createContext<LocaleContextProps | undefined>(undefined);

export default LocaleContext;
