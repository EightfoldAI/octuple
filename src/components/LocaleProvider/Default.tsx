import locale from '../Locale/Default';

export default locale;
