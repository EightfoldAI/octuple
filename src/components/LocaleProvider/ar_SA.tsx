import locale from '../Locale/ar_SA';

export default locale;
