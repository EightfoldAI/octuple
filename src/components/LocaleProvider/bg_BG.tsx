import locale from '../Locale/bg_BG';

export default locale;
