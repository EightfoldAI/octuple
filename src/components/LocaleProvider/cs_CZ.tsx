import locale from '../Locale/cs_CZ';

export default locale;
