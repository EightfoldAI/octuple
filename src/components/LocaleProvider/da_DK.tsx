import locale from '../Locale/da_DK';

export default locale;
