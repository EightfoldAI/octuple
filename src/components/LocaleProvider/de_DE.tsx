import locale from '../Locale/de_DE';

export default locale;
