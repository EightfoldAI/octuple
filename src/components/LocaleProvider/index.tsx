import React from 'react';
import memoizeOne from 'memoize-one';
import type { PickerLocale as DatePickerLocale } from '../DateTimePicker/DatePicker/Generate/Generate.types';
import type { ValidateMessages } from '../Form/Internal/OcForm.types';
import LocaleContext from './Context';

export interface Locale {
    locale: string;
    DatePicker?: DatePickerLocale;
    TimePicker?: Record<string, any>;
    global?: Record<string, any>;
    Form?: {
        optional?: string;
        defaultValidateMessages: ValidateMessages;
    };
}

export interface LocaleProviderProps {
    locale: Locale;
    children?: React.ReactNode;
}

export default class LocaleProvider extends React.Component<
    LocaleProviderProps,
    any
> {
    static defaultProps = {
        locale: {},
    };

    constructor(props: LocaleProviderProps) {
        super(props);
    }

    getMemoizedContextValue = memoizeOne(
        (localeValue: Locale): Locale & { exist?: boolean } => ({
            ...localeValue,
            exist: true,
        })
    );

    render() {
        const { locale, children } = this.props;
        const contextValue = this.getMemoizedContextValue(locale);
        return (
            <LocaleContext.Provider value={contextValue}>
                {children}
            </LocaleContext.Provider>
        );
    }
}
