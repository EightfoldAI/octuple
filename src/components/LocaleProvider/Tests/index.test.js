/* eslint-disable react/no-multi-comp */
import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import MatchMediaMock from 'jest-matchmedia-mock';
import MockDate from 'mockdate';
import dayjs from 'dayjs';
import LocaleProvider from '../LocaleReceiver';
import { DatePicker, RangePicker } from '../../DateTimePicker/DatePicker';
import { TimePicker } from '../../DateTimePicker/TimePicker/TimePicker';
import { Modal } from '../../Modal';
import { Pagination } from '../../Pagination';
import { Select } from '../../Select';
import { Table } from '../../Table';
import arSA from '../ar_SA';
import bgBG from '../bg_BG';
import csCZ from '../cs_CZ';
import daDK from '../da_DK';
import deDE from '../de_DE';
import elGR from '../el_GR';
import enGB from '../en_GB';
import enUS from '../en_US';
import esES from '../es_ES';
import esDO from '../es_DO';
import esMX from '../es_MX';
import fiFI from '../fi_FI';
import frBE from '../fr_BE';
import frCA from '../fr_CA';
import frFR from '../fr_FR';
import heIL from '../he_IL';
import hiIN from '../hi_IN';
import hrHR from '../hr_HR';
import htHT from '../ht_HT';
import huHU from '../hu_HU';
import itIT from '../it_IT';
import jaJP from '../ja_JP';
import koKR from '../ko_KR';
import msMY from '../ms_MY';
import nbNO from '../nb_NO';
import nlBE from '../nl_BE';
import nlNL from '../nl_NL';
import plPL from '../pl_PL';
import ptBR from '../pt_BR';
import ptPT from '../pt_PT';
import roRO from '../ro_RO';
import ruRU from '../ru_RU';
import skSK from '../sk_SK';
import srRS from '../sr_RS';
import svSE from '../sv_SE';
import teIN from '../te_IN';
import thTH from '../th_TH';
import trTR from '../tr_TR';
import ukUA from '../uk_UA';
import viVN from '../vi_VN';
import zhCN from '../zh_CN';
import zhTW from '../zh_TW';

Enzyme.configure({ adapter: new Adapter() });

let matchMedia;

const locales = [
  arSA,
  bgBG,
  csCZ,
  daDK,
  deDE,
  elGR,
  enGB,
  enUS,
  esES,
  esDO,
  esMX,
  fiFI,
  frBE,
  frCA,
  frFR,
  heIL,
  hiIN,
  hrHR,
  htHT,
  huHU,
  itIT,
  jaJP,
  koKR,
  msMY,
  nbNO,
  nlBE,
  nlNL,
  plPL,
  ptBR,
  ptPT,
  roRO,
  ruRU,
  skSK,
  srRS,
  svSE,
  teIN,
  thTH,
  trTR,
  ukUA,
  viVN,
  zhCN,
  zhTW,
];

const columns = [
  {
    title: 'Name',
    dataIndex: 'name',
    filters: [
      {
        text: 'filter1',
        value: 'filter1',
      },
    ],
  },
  {
    title: 'Age',
    dataIndex: 'age',
  },
];

const App = () => (
  <div>
    <Pagination defaultCurrent={1} total={50} />
    <Select
      options={[
        {
          text: 'Mia',
          value: 'mia',
        },
        {
          text: 'Lola',
          value: 'lola',
        },
      ]}
      style={{ width: 200 }}
    />
    <DatePicker />
    <TimePicker defaultOpenValue={dayjs()} />
    <RangePicker style={{ width: 200 }} />
    <Table dataSource={[]} columns={columns} />
    <Modal title={'Locale Modal'} visible body={<p>Locale Modal</p>} />
  </div>
);

describe('Locale Provider', () => {
  beforeAll(() => {
    matchMedia = new MatchMediaMock();
    MockDate.set(dayjs('2017-09-18T03:30:07.795').valueOf());
  });

  afterEach(() => {
    matchMedia.clear();
  });

  afterAll(() => {
    MockDate.reset();
  });

  locales.forEach((locale) => {
    test(`should display the text as ${locale.locale}`, () => {
      const Test = () => (_, __) => {
        <LocaleProvider locale={locale}>
          <App />
        </LocaleProvider>;
      };
      const wrapper = mount(<Test />);
      expect(wrapper.render()).toMatchSnapshot();
    });
  });

  test('set dayjs locale when locale changes', () => {
    const Test =
      ({ locale }) =>
      (_, __) => {
        <LocaleProvider locale={locale}>
          <div>
            <DatePicker defaultValue={dayjs()} />
          </div>
        </LocaleProvider>;
      };

    const wrapper = mount(<Test />);
    wrapper.setProps({ locale: zhCN });
    expect(wrapper.render()).toMatchSnapshot();
    wrapper.setProps({ locale: frFR });
    expect(wrapper.render()).toMatchSnapshot();
    wrapper.setProps({ locale: null });
    expect(wrapper.render()).toMatchSnapshot();
  });
});
