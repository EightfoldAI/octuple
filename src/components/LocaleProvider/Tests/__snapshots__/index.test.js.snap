// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Locale Provider set dayjs locale when locale changes 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider set dayjs locale when locale changes 2`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider set dayjs locale when locale changes 3`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as ar 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as bg 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as cs 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as da 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as de 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as el 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as en 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as en-gb 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as es 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as es-do 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as es-mx 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as fi 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as fr 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as fr 2`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as fr 3`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as fr 4`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as he 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as hi 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as hr 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as hu 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as it 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as ja 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as ko 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as ms-my 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as nb 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as nl 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as nl-be 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as pl 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as pt 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as pt-br 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as ro 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as ru 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as sk 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as sr 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as sv 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as te 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as th 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as tr 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as uk 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as vi 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as zh-cn 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Locale Provider should display the text as zh-tw 1`] = `
LoadedCheerio {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
